DigitalLearningSolutions.Data/Services/CourseContentService.cs

```diff
@@ -80,18 +80,55 @@ GROUP BY
             ).FirstOrDefault();
         }
 
-        public int GetProgressId(int candidateId, int customisationId)
+        public int? GetOrCreateProgressId(int candidateId, int customisationId, int centreId)
         {
-            // TODO HEEDLS-202: change QueryFirstOrDefault to creating progress record if not found
-            return connection.QueryFirstOrDefault<int>(
-                @"SELECT ProgressId
-                        FROM Progress
-                        WHERE CandidateID = @candidateId
-                          AND CustomisationID = @customisationId",
-                new { candidateId, customisationId }
+            if (DoesProgressExist(candidateId, customisationId))

```

Comment:
You don't really need this as a separate check. You can just get the progress id and check if it's null 🙂